h2o-persist-s3/src/main/java/water/persist/PersistS3.java

```diff
@@ -488,8 +490,8 @@ protected String wrapKey(String s) {
   }
 
 
-  Cache _bucketCache = new Cache();
-  HashMap<String, KeyCache> _keyCaches = new HashMap<>();
+  static volatile Cache _bucketCache = new Cache();
```

Comment:
This was here for eternity and was accidentally removed for no reason (probably as a leftover) in of my PRs and nobody noticed. Returned it back.